pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -289,6 +289,7 @@ def simulate_and_vjp(
         "QFT",
         "ECR",
         "BlockEncode",
+        "C(BlockEncode)",
```

Comment:
Have we made sure we are properly hitting this [line](https://github.com/PennyLaneAI/pennylane-lightning/blob/maa/ctrl-blockencode/pennylane_lightning/lightning_qubit/_state_vector.py#L244)? If not, we'll hit this [line](https://github.com/PennyLaneAI/pennylane-lightning/blob/maa/ctrl-blockencode/pennylane_lightning/lightning_qubit/_state_vector.py#L313) which is prohibitively expensive when there are many control wires. We could also add a test with, say, 20 control wires which will blow up the memory if we are not using the correct kernel.

Reply:
Thanks @vincentmr! Yes, we are hitting the `operations` and this line :+1: